### ISSUE TITLE
docparams does not recognize the optional return value name numpydoc secification

### ISSUE THREAD

**Issue by youngmit** [PyCQA/pylint#2030]
### Steps to reproduce
1. Run pylint over
```python
def do_a_thing(field_value):
    """Does something

    Does some stuff

    Returns
    -------
    thing : dict
        some junk in a dict
    """
    thing = {"a": field_value, "b": field_value}
    return thing
```
with docparams enabled.

### Current behavior
Get warning: `W:  2, 0: Missing return type documentation (missing-return-type-doc)`

### Expected behavior
No warning

### pylint --version output
```
C:\Users\bleep\tmp>pylint --version
No config file found, using default configuration
pylint 1.8.4,
astroid 1.6.3
Python 3.6.4 (v3.6.4:d48eceb, Dec 19 2017, 06:54:40) [MSC v.1900 64 bit (AMD64)]
```

It looks like pylint isn't honoring the optional return value names described in the numpydoc spec (https://numpydoc.readthedocs.io/en/latest/format.html#sections). Looking at the offending RE, we have:
```python
re_returns_line = re.compile(r"""
        \s* ({type})$ # type declaration
        \s* (.*)                       # optional description
    """.format(
        type=GoogleDocstring.re_multiple_type,
    ), re.X | re.S | re.M)
``` 

I am happy to toss together a PR on this, just wanted to make sure this isn't somehow a PEBCAK.
